openedx/core/djangoapps/user_api/tests/test_views.py

```diff
@@ -1062,6 +1062,39 @@ def test_register_form_default_fields(self):
             }
         )
 
+    @override_settings(PASSWORD_COMPLEXITY={'NON ASCII': 1, 'UPPER': 3})
+    def test_register_form_password_complexity(self):
+        no_extra_fields_setting = {}
+
+        # Without enabling password policy
+        self._assert_reg_field(
+            no_extra_fields_setting,
+            {
+                u'name': u'password',
+                u'label': u'Password',
+                u'restrictions': {
+                    'min_length': password_min_length(),
+                    'max_length': password_max_length(),
+                },
+            }
+        )
+
+        # Now with an enabled password policy
+        with mock.patch.dict(settings.FEATURES, {'ENFORCE_PASSWORD_POLICY': True}):
+            self._assert_reg_field(
+                no_extra_fields_setting,
+                {
+                    u'name': u'password',
+                    u'label': u'Password',
+                    u'restrictions': {
+                        'min_length': password_min_length(),
+                        'max_length': password_max_length(),
+                        'non_ascii': 1,
+                        'upper': 3,
```

Comment:
Are these all built into Django?

Reply:
What do you mean?

Reply:
Sorry, the functionality for validating these restrictions is built into Django, or is something we wrote?

Reply:
Something we wrote. The only people that check it right now are mobile I think. They don't validate these new values yet. This is just to enable them to do so at their own release pace.